### PR TITLE
Add a kfdef that deploys latest dspo.

### PR DESCRIPTION
**Pull request by HumairAK**
## Description
Add a sample kfdef with instructions to deploy latest dspo in this repo from main branch.

## How Has This Been Tested?
Install ODH, follow readme instructions in pr.

## Merge criteria:
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: kfdef/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+```bash 
+git clone https://github.com/opendatahub-io/data-science-pipelines-operator.git 
+cd data-science-pipelines-operator/kfdef 
+
+# If this namespace does not exist
+oc new-project odh-applications 
+
+# Then run
+oc -n odh-applications apply -f kfdef.yaml
```
Comment (accorvin) [inline kfdef/README.md:25]: I recently learned that you can pass a URL to `oc apply`, so this command could be:
```
oc apply -f https://raw.githubusercontent.com/HumairAK/data-science-pipelines-operator/add_kfdef/kfdef/kfdef.yaml -n odh-applications
```

If we document it that way then we can eliminate the `git clone` and `cd` steps above.

Comment (HumairAK) [reply kfdef/README.md]: done

File: kfdef/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Deploy Kfdef
+
+Clone this repository then run the following commands: 
+
+```bash 
+# If this namespace does not exist
+oc new-project odh-applications 
+
+# Then run
+oc apply -f https://raw.githubusercontent.com/opendatahub-io/data-science-pipelines-operator/main/kfdef/kfdef.yaml -n odh-applications
```
Comment (DharmitD) [inline kfdef/README.md:22]: I'm testing the kfdef install, is this url incorrect? I see a 404.

`error: unable to read URL "https://raw.githubusercontent.com/opendatahub-io/data-science-pipelines-operator/main/kfdef/kfdef.yaml", server reported 404 Not Found, status code=404`

**Review by DharmitD — COMMENTED** [review]
/lgtm

Tested out by applying the kfdef, DSPO comes up correctly.

**Pull request merged**